hydrus/resources.py

```diff
@@ -144,15 +145,17 @@ def post(self, id_: str, path: str) -> Response:
         if checkClassOp(class_type, "POST") and check_writeable_props(class_type, object_):
             # Check if class_type supports POST operation
             obj_type = getType(class_type, "POST")
+            link_props = get_link_props(class_type, object_)
             # Load new object and type
             if validObject(object_) and object_["@type"] == obj_type and check_required_props(
-                    class_type, object_):
+                    class_type, object_) and isinstance(link_props, Dict):
```

Comment:
Since get_link_props always return a dict, is `isinstance(link_props, Dict)` check necessary?

Reply:
It returns False when type doesn't match. 
Class_type of resource provided as link should match to class_type defined as range of the link_property in the ApiDoc.

Reply:
Removed that and made an explicit check for type-checking of links.